controllers/drplacementcontrol_controller.go

```diff
@@ -495,9 +526,16 @@ func (r *DRPlacementControlReconciler) reconcileDRPCInstance(ctx context.Context
 		return ctrl.Result{RequeueAfter: time.Second * initialWaitTime}, nil
 	}
 
+	vrgs, err := r.getVRGsFromManagedClusters(drpc, drPolicy)
```

Comment:
Retrieving the VRGs from Managed clusters before creating DRPC instance is appealing, however, one of the problems that we will face is; what should we do in a case of a failure to retrieve a VRG from one of the MCs. Should we just fail the reconciliation for that DRPC? Probably this call should move to initilaDeployment()/runFailover()/runRelocation and the failure will be handled case by case. For now, it works fine.

Reply:
In a failover case, one of the clusters would be unavailable, and so it wont progress beyond here (as it would fail to get the VRG).

The question is though, will we get an older copy of the VRG from the existing MCV that was created to monitor the same. If so, then we would make progress, as that would report "something" based on which we would make further progress.

Reply:
That's right, that why I put that comment.  At the end of that comment, I said that it works fine because somehow I assumed that trying to get a VRG from a failed cluster will return MCV condition type `ConditionViewProcessing` and the condition reason `ReasonGetResourceFailed`, which we interpret as `NotFound` response.  But I think that's a bad assumption.  It is safe to just move that call to each action and for the failover, we skip the failed cluster on error.

Reply:
Was testing MCV creation post cluster down, and it returns no status.

As an aside, an already created MCV returns a prior status (or last known status) when the cluster is down.

Was thinking about dealing with managed cluster conditions `ManagedClusterConditionAvailable` but that is too big a change just as yet.

Continuing to review under the basis that we will,
- Fetch VRG from both managedclusters for `deploy` and `relocate` cases, and keep reconciling otherwise
    - There maybe corner cases if we get a VRG that is primary on the surviving cluster (but will think through that later)
- Fetch VRG from the surviving cluster for `failover` cases